source/TestContext.h

```diff
@@ -28,13 +28,24 @@ friend class Test;
 
 
 private:
-	// Pointer to the test we are running.
-	std::vector<const Test *> testToRun;
+	// Struct to describe a running test and running test-step within the test.
+	struct ActiveTestStep{
+		const Test *test;
+		unsigned int step;
+
+		// Support operators for usage in containers like map and set.
+		bool operator==(const ActiveTestStep &rhs) const;
+		bool operator<(const ActiveTestStep &rhs) const;
+		bool operator>(const ActiveTestStep &rhs) const;
+	};
+
+private:
+	// Reference to the currently running test and test-step within the test.
+	std::vector<ActiveTestStep> callstack;
 
 	// Teststep to run.
-	std::vector<unsigned int> stepToRun = { 0 };
```

Comment:
There's an initial element here that isn't present in `callstack`. I don't know whether this was intentional or not, so I'm just mentioning it.

Reply:
Yes, this is not an issue; in the original code only the testToRun variable to initialized with a test (and we needed stepsToRun to always start with step 0); in the new code we initialize the callstack with the test and a 0 for the step, so the step also does get initialized there correctly.